target/linux/pistachio/base-files/etc/init.d/bt-config

```diff
@@ -0,0 +1,54 @@
+#!/bin/sh /etc/rc.common
+#
+# Copyright (C) 2017 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+START=85
+STOP=85
+
+config_bluetooth() {
+	. /lib/functions/mac.sh
+	. /lib/functions/bluetooth.sh
+
+	local IS_BT_RUNNING
+	local timeout=10
+	local time=`awk -F "." '{print $1}' /proc/uptime | awk '{print $1}'`
+	local elap_time=0
+
+	while [ -z "$IS_BT_RUNNING" ]
+	do
+		if [ $elap_time -lt $timeout ]
+		then
+			elap_time=$( elapsed_time $time )
+		else
+			logger " waiting for bluetoothd daemon: bt-config timed out "
+			return 1
+		fi 
+		IS_BT_RUNNING=`ps | grep bluetoothd | grep -v grep`
+	done
+	# give some time for the bluetoothd to come up before applying 
```

Comment:
In stead of waiting for "some" random time, can't we bank on some event for applying configuration?